### PR TITLE
Add missing exception checkpoint

### LINKED ISSUE #38826: CoreCLR tests failed with Mono interpreter: Interop/NativeLibrary/API/NativeLibraryTests
**Repro:** 
1. cd src/mono/netcore
2. export MONO_ENV_OPTIONS=--interpreter
3. make run-tests-coreclr CoreClrTest="bash ../../artifacts/tests/coreclr/OSX.x64.Debug/Interop/NativeLibrary/API/NativeLibraryTests/NativeLibraryTests.sh"

**Log:**
```
  BEGIN EXECUTION
  /Users/fanyang/Documents/work/dotnet_test/runtime/artifacts/tests/coreclr/OSX.x64.Debug/Tests/Core_Root/corerun NativeLibraryTests.dll ''
  Unhandled exception System.DllNotFoundException: /Users/fanyang/Documents/work/dotnet_test/runtime/artifacts/tests/coreclr/OSX.x64.Debug/Interop/NativeLibrary/API/NativeLibraryTests/notfound
     at System.String.Format(String format, Object arg0, Object arg1) in /Users/fanyang/Documents/work/dotnet_test/runtime/src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs:line 475
     at NativeLibraryTest.EXPECT(TestResult actualValue, TestResult expectedValue) in /Users/fanyang/Documents/work/dotnet_test/runtime/src/coreclr/tests/src/Interop/NativeLibrary/API/NativeLibraryTests.cs:line 206
     at NativeLibraryTest.Main() in /Users/fanyang/Documents/work/dotnet_test/runtime/src/coreclr/tests/src/Interop/NativeLibrary/API/NativeLibraryTests.cs:line 78
  Expected: 100
  Actual: 156
  END EXECUTION - FAILED
```

### PR DESCRIPTION
**Pull request by fanyang-mono**
Fixes #38826

### CODE REVIEW

**Comment by SamMonoRT** [comment]
Enable the tests (remove from issues.target file) to make sure change actually fixes the affected ones in #38826

**Comment by fanyang-mono** [comment]
> Enable the tests (remove from issues.target file) to make sure change actually fixes the affected ones in #38826

Oops, good catch!

File: src/mono/mono/mini/interp/interp.c [hunk 1]
```diff
@@ -3847,6 +3847,7 @@ interp_exec_method (InterpFrame *frame, ThreadContext *context, FrameClauseArgs
 			frame->state.ip = ip + 6;
 			ves_pinvoke_method (csignature, (MonoFuncV)code, context, frame, &retval, save_last_error, cache, sp);
 
+			EXCEPTION_CHECKPOINT_GC_UNSAFE;
```
Comment (CoffeeFlux) [inline src/mono/mono/mini/interp/interp.c:3850]: Aren't we back in GC safe here?

Comment (fanyang-mono) [reply src/mono/mono/mini/interp/interp.c:3850]: Followed the same pattern we did for `MINT_CALLI_NAT_FAST`.

Comment (CoffeeFlux) [reply src/mono/mono/mini/interp/interp.c:3850]: It's not clear to me why we're in GC unsafe there, though I guess if it throws we must be? @BrzVlad do you know what's going on here?

Comment (fanyang-mono) [reply src/mono/mono/mini/interp/interp.c:3850]: Here is the story behind it https://github.com/mono/mono/pull/17016. You could also read the issue it mentioned in this PR. In short, usually the interpreter operates in GC Unsafe state.

**Review by SamMonoRT — APPROVED** [review]
Please check the logs to verify if all the Interop/NativeLibrary/API/NativeLibraryTests were included and passed prior to merging the PR.

**Review by BrzVlad — APPROVED** [review]

**Pull request merged**